src/ui/components/Location/component.ts

```diff
@@ -171,31 +170,55 @@ export default class LocationComponent extends Component {
   async loadFromAPI(locationSignature: RecordSignature) {
     let { store } = this.args;
     try {
-      let location: Location = await store.query((q) =>
+      this.location = await store.query((q) =>
         q.findRecord(locationSignature)
       );
-      let measurements: Measurement[] = await store.query((q) =>
+    } catch {
+      // Fail silently
+    }
+    try {
+      // Remove old data
+      let cachedResults = this.measurements = store.cache.query((q) =>
         q.findRelatedRecords(locationSignature, 'measurements')
       );
+      cachedResults.forEach((oldMeasurement: Measurement) => {
+      let measurementSignature = { type: 'measurement', id: oldMeasurement.id };
+      store.update((t) => t.removeFromRelatedRecords(
+        locationSignature,
+        'measurements',
+        measurementSignature
+      ));
+      store.update((t) =>
+        t.removeRecord(measurementSignature));
+      });
 
-      if (location && measurements) {
-        this.location = location;
-        this.measurements = measurements;
+      // Fetch new data
+      this.measurements = await store.query((q) =>
+        q.findRelatedRecords(locationSignature, 'measurements')
+      );
 
-        // Remember we saw this location
-        let currentDate = new Date().toISOString();
-        store.update((t) =>
-          t.replaceAttribute(locationSignature, 'visitedAt', currentDate)
-        );
-      }
+      // Remember we saw this location
+      let currentDate = new Date().toISOString();
+      store.update((t) =>
+        t.replaceAttribute(locationSignature, 'visitedAt', currentDate)
+      );
     } catch {
       // Only show error if no records could be found in any source
       this.notFound = !this.recordsFound;
+
+      // Restore measurements to cache if any
```

Comment:
I wonder why this is necessary to do manually now but wasn't before apparently?

Reply:
Now we explicitly delete all the local measurements before knowing if we'll get a successful result from the API. We restore them in case fetching the measurements fail. In the past, we never deleted records and we just kept adding the new ones to the store.